accessor-alias/sycl-2.2/index.md

```diff
@@ -0,0 +1,686 @@
+is because this is the default access mode,
+which enables writing `accessor<const dataT>` in the above example.
+The SYCL scheduler treats this combination
+as if the access mode was `access::mode::read`.
+
+### Subscript operator
+
+After _requesting_ access to data by constructing an accessor,
+`accessor::operator[]` is the function that enables
+the second main functionality of the accessor class:
+it _provides_ access to data.
+
+We propose that `accessor::operator[]` returns `reference_t`,
+which is defined as:
+
+* `const T&` when the access mode is `access::mode::read`
+  or when `dataT` is `const`-qualified
+* `T&` otherwise
+
+### Implicit conversions
+
+In order to simplify user code,
+we propose allowing certain implicit conversions
+that don't modify scheduling information.
+
+In standard C++ code, adding `const` qualifiers is almost always allowed.
+In SYCL, going from a `read_write` accessor to a `read` accessor
+is analogous to adding the `const` qualifier.
+This allows us to have the following rules regarding `const`:
+
+1. Convert an accessor of `dataT` data type
+   to an accessor of `const dataT` data type,
+   all other template parameters being equal
+1. Convert an accessor of `const dataT` data type
+   and `access::mode::read_write` mode
+   to an accessor of `dataT` and `access::mode::read` mode,
+   all other template parameters being equal
+1. Convert an accessor of `const dataT` data type
+   and `access::mode::read_write` mode
+   to an accessor of `const dataT` and `access::mode::read` mode,
+   all other template parameters being equal
+1. A combination of the previous rules,
+   allow any of these accessor combinations to be implicitly convertible
+   between each other:
+    * {`dataT`, `read`}
+    * {`const dataT`, `read`}
+    * {`const dataT`, `read_write`}
+
+An example of code these rules enable:
+
+```cpp
+buffer<int> buf{...};
+
+void const_int_read_write(accessor<const int> acc) {...}
+void int_read(accessor<int, 1, access::mode::read> acc) {...}
+void const_int_read(accessor<const int, 1, access::mode::read> acc) {...}
+
+q.submit([&](handler& cgh){
+    auto accInt =
+        buf.get_access(cgh); // accessor<int, 1, read_write>
+    auto accIntConst =
+        accessor<const int>{buf, cgh}; // accessor<const int, 1, read_write>
+
+    // 1
+    // accInt requested `read_write` mode,
+    // the scheduler may or may not be able to optimize this
+    const_int_read_write(accInt);
+
+    // 2
+    // Scheduler treats the combination of `const int` and `read_write` mode
+    // the same as if `read` mode was used,
+    // so this code is optimal
+    int_read(accIntConst);
+
+    // 3
+    // Similar to rule 2
+    const_int_read(accIntConst);
+
+    // 4
+    // Allow all combinations
+    const_int_read_write(accIntConst)
+    int_read(accInt)
+    const_int_read(accInt)
+
+    ...
+});
+```
+
+We also propose the following rules for simplifying the access mode:
```

Comment:
It could be worth making the `discard` part of the `access::mode` a separate runtime parameter or property?

Reply:
I was considering that, but it seems like the existing access modes work reasonably well with this proposal, so a new tag might actually be adding verbosity.

Reply:
I've added `property::discard` and `property::discard_v`